test/common/upstream/cds_api_impl_test.cc

```diff
@@ -241,6 +291,141 @@ TEST_F(CdsApiImplTest, Basic) {
   EXPECT_EQ(13237225503670494420U, store_.gauge("cluster_manager.cds.version").value());
 }
 
+TEST_F(CdsApiImplTest, CdsPauseOnWarming) {
+  interval_timer_ = new Event::MockTimer(&dispatcher_);
+  InSequence s;
+
+  setup();
+
+  // Two clusters updated, both warmed up.
+  const std::string response1_json = R"EOF(
+{
+  "version_info": "0",
+  "resources": [
```

Comment:
Generally prefer YAML for new test fragments.

Reply:
Introduced `parseResponseMessageFromYaml()` and converted all JSON messages in the unit test to YAML (including the old ones).

Reply:
Thanks, appreciated.